daemons/dss-checkout-sfn/app.py

```diff
@@ -0,0 +1,96 @@
+import os
+import sys
+
+import domovoi
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), 'domovoilib'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from dss.util.state_machine.checkout_states import state_machine_def
+from dss.util.email import send_checkout_success_email, send_checkout_failure_email
+from dss.util.checkout import (parallel_copy, get_dst_bundle_prefix, get_manifest_files,
+                               validate_file_dst, validate)
+
+app = domovoi.Domovoi()
+dss.Config.set_config(dss.BucketConfig.NORMAL)
+
+replica = "aws"
+
+logger = dss.get_logger()
+email_sender = dss.Config.get_notification_email()
+default_checkout_bucket = dss.Config.get_s3_checkout_bucket()
+
+@app.step_function_task(state_name="ScheduleCopy", state_machine_definition=state_machine_def)
+def worker(event, context):
+    bundle_id = event["bundle"]
+    version = event["version"]
+    dss_bucket = event["dss_bucket"]
+    dst_bucket = get_dst_bucket(event)
+
+    scheduled = 0
```

Comment:
Might be preferable to just call `len()` on the results of `get_manifest_files(..)` to get scheduled.

I realize that `get_manifest_files(..)` returns a generator, but I think that's almost certainly premature optimization.  We are retrieving the entire manifest into memory.

Reply:
I'd like to keep the daemon code as slim as possible. get_manifest_files returns a generator and I wanted to make sure I can unit test that code.